osquery/tables/system/posix/ssl_cert.cpp

```diff
@@ -0,0 +1,199 @@
+  r["issuer_organization"] = std::string(temp);
+
+  X509_NAME_get_text_by_NID(
+      issuer_name, NID_organizationalUnitName, temp, SHORT_STR);
+  r["issuer_organization_unit"] = std::string(temp);
+
+  // set period of validity
+  ASN1_TIME* valid_from = X509_get_notBefore(cert);
+  ASN1_TIME* valid_to = X509_get_notAfter(cert);
+  BIO* b = BIO_new(BIO_s_mem());
+
+  ASN1_TIME_print(b, valid_from);
+  BIO_gets(b, temp, SHORT_STR);
+  r["valid_from"] = std::string(temp);
+
+  ASN1_TIME_print(b, valid_to);
+  BIO_gets(b, temp, SHORT_STR);
+  r["valid_to"] = std::string(temp);
+  BIO_free(b);
+
+  // set sha 256 & 1 fingerprint
+  EVP_MD* digest = (EVP_MD*)EVP_sha256();
+  unsigned len = SHORT_STR;
+  X509_digest(cert, digest, (unsigned char*)temp, &len);
+  char sha_text[SHORT_STR + 1];
+  memset(sha_text, 0, sizeof(sha_text));
+
+  for (unsigned i = 0; i < len; i++) {
+    char byt[5];
+    sprintf(byt, "%02X%c", temp[i], (i + 1 == len) ? '\0' : ':');
+    strcat(sha_text, byt);
+  }
+
+  r["sha256_fingerprint"] = std::string(sha_text);
+  digest = (EVP_MD*)EVP_sha1();
+  X509_digest(cert, digest, (unsigned char*)temp, &len);
+  memset(sha_text, 0, sizeof(sha_text));
+
+  for (unsigned i = 0; i < len; i++) {
+    char byt[5];
+    sprintf(byt, "%02X%c", temp[i], (i + 1 == len) ? '\0' : ':');
+    strcat(sha_text, byt);
+  }
+
+  r["sha1_fingerprint"] = std::string(sha_text);
+
+  return;
+}
+
+static void getSslCert(const char* issued_cname, QueryData& results) {
+  char* def_cert_dir;
+
+  // common certificate directory for POSIX
+  const std::string cert_dir_list[6] = {"/etc/pki/tls/certs",
+                                        "/etc/ssl/certs",
+                                        "/etc/ssl",
+                                        "/etc/pki/ca-trust/extracted/pem",
+                                        "/usr/local/share/certs",
+                                        "/etc/openssl/certs"};
+
+  def_cert_dir = getenv(X509_get_default_cert_dir_env());
+
+  if (!def_cert_dir) {
+    def_cert_dir = (char*)X509_get_default_cert_dir();
+  }
+
+  fs::path full_path(fs::initial_path<fs::path>());
+
+  full_path = fs::system_complete(fs::path(def_cert_dir));
+
+  // check if default certificate directory is empty
+  if (fs::is_empty(full_path)) {
+    // use the common certificate directory
+    for (unsigned i = 0;
+         i < (unsigned)(sizeof(cert_dir_list) / sizeof(cert_dir_list[0]));
+         i++) {
+      fs::path temp_path = fs::system_complete(fs::path(cert_dir_list[i]));
+
+      if (fs::exists(temp_path) && !fs::is_empty(temp_path)) {
+        def_cert_dir = (char*)cert_dir_list[i].c_str();
+        full_path = temp_path;
+        break;
+      }
+    }
+  }
+
+  if (fs::is_directory(full_path)) {
+    fs::directory_iterator end_iter;
+
+    FILE* fp;
+    // parse each certificate in the default directory
+    for (fs::directory_iterator dir_itr(full_path); dir_itr != end_iter;
```

Comment:
Rather than walking through your list above looking for directories, maybe you could change this logic below to something like how the Darwin `crashes` table [iterates over all crash logs in a directory](https://github.com/facebook/osquery/blob/master/osquery/tables/system/darwin/crashes.cpp#L161)? This should allow you to walk over every cert in a container of your specifying.